opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchIndexScan.java

```diff
@@ -99,11 +102,12 @@ public ExprValue next() {
     return iterator.next();
   }
 
-  private void fetchNextBatch() {
+  protected OpenSearchResponse fetchNextBatch() {
     OpenSearchResponse response = client.search(request);
     if (!response.isEmpty()) {
       iterator = response.iterator();
     }
+    return response;
```

Comment:
instead of returning here, could you just set the rawResponse directly?

Reply:
Changed in https://github.com/Bit-Quill/opensearch-project-sql/commit/c7208ea1f35ee9d657fb6898e800edb77e142ab3

Reply:
Correction, I had to revert because this function is being called in `open()` and `hasNext()` and we only want to set `rawResponse` when `open()` is called.